ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentImpl.java

```diff
@@ -390,10 +428,53 @@ private void hideCameraTriggerButtonAnimated() {
         mButtonCameraTrigger.setEnabled(false);
     }
 
-    private void showNoPermissionView() {
-        hideCameraPreviewAnimated();
+    @Override
+    public void showInterface() {
+        if (isNoPermissionViewVisible()) {
+            return;
+        }
+        showInterfaceAnimated();
+    }
+
+    private void showInterfaceAnimated() {
+        showCameraTriggerButtonAnimated();
+        showDocumentCornerGuidesAnimated();
+        if (mImportDocumentButtonEnabled) {
+            showImportDocumentButtonAnimated();
+        }
+    }
+
+    private void showImportDocumentButtonAnimated() {
+        LOG.debug("Showing document import button");
+        mButtonImportDocument.animate().alpha(1.0f);
+        mButtonImportDocument.setEnabled(true);
+    }
+
+    @Override
+    public void hideInterface() {
+        if (isNoPermissionViewVisible()) {
+            return;
+        }
+        hideInterfaceAnimated();
+    }
+
+    private void hideInterfaceAnimated() {
         hideCameraTriggerButtonAnimated();
         hideDocumentCornerGuidesAnimated();
+        if (mImportDocumentButtonEnabled) {
+            hideImportDocumentButtonAnimated();
+        }
+    }
+
+    private void hideImportDocumentButtonAnimated() {
+        LOG.debug("Hiding document import button");
```

Comment:
Same as above, if logging is really necessary at this point this might be `verbose`

Reply:
Yeah, verbose sounds good. 👍